packages/KPrompt/KPrompt.vue

```diff
@@ -144,6 +159,10 @@ export default {
       return this.capitalize(this.type)
     },
     disableProceedButton () {
+      if (this.actionPending) {
+        return true
+      }
```

Comment:
When the Proceed button is disabled, is your thought that they clicked the button to proceed and it's finishing a process before closing the prompt?

If yes, I believe you should disable the cancel button, `esc` press, and close button click to prevent them from closing the prompt before the process finishes as well.

Reply:
I had thought about this but we don't currently do this anywhere else in the UI. I think we are only trying to prevent spam clicking, not closing the prompt.

Reply:
The action completes in the background. We don't want to interrupt or cancel the call at that stage.

Reply:
That's why I'm thinking we would disable everything. If the user closes the prompt before it completes, we will then be emitting 2 events back to the caller, `canceled` and `proceed` (assuming the `proceed` will still be called after the action completes). 

If the consuming app has logic to respond to both events, we could potentially be giving them a "false cancel" even though the user actually did choose to proceed. 

Actually, now I'm curious -- how can the consuming UI ever set `actionPending`? The user clicks the `proceed` button, and immediately the modal is closed unless the `actionPending` prop is already somehow set to true?

Reply:
^ more on this (commenting here since you're in meetings all day)

> This boolean indicates if an action is being taken on the dialog and we should disable the action button to prevent spam clicking.

Since the consuming app is only notified when they click the button (not before) how would this prop get switched to true to show the pending state on click?

Reply:
This is interesting, so the typical implementation I am expecting is there will be state transitions happening in the submit button's action handler like:

![image](https://user-images.githubusercontent.com/67973710/155235948-23138e49-7d9e-4f55-a44e-b5d2c578d7c9.png)

`actionPending` is determined by this state:

![image](https://user-images.githubusercontent.com/67973710/155236031-3dcfcd81-0e89-41e9-848f-0827e306850f.png)

Also, typically it is ok if both `proceed` and `cancel` events are triggered because in general the only thing happening in a cancel is 
- the dialog is closed
- form values are cleared
- we may send a state change event (which would be handled correctly because it's a no-op if sending `CANCEL` event when in `pending` state, we only take action `RESOLVE` or `REJECT` events from that state)

State machine:

![image](https://user-images.githubusercontent.com/67973710/155236390-f8178f78-74dc-462a-bac4-1b8a35cbf4a6.png)

Reply:
Thanks for the context. 

So based on your example, the `@proceed` emitted event doesn't actually close the modal -- the user has to handle that themselves, right? I was missing that piece; I was thinking clicking the proceed button also closed the prompt when I read through this at the end of the day.